python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -166,6 +166,90 @@ def test_library_management_path_prefix(moto_s3_uri_incl_bucket, boto_client):
+
+    assert results_dict[0].metadata == {"meta1": 6}
+    assert results_dict[1].metadata == {"meta2": 7}
+    assert results_dict[2].metadata is None
+    assert results_dict[3].metadata == {"meta3": 8}
+
+    lib.delete("sym1", versions=2)
+    lib.delete("sym2", versions=2)
+    lib.delete("sym3", versions=2)
+
+    results_dict = lib.read_metadata_batch(["sym1", "sym2", "sym_no_exist" ,"sym3"])
+    assert results_dict[0].metadata == {"meta1": 4}
+    assert results_dict[1].metadata == {"meta2": 5}
+    assert results_dict[2].metadata is None
+    assert results_dict[3].metadata == {"meta3": 6}
+
+    assert lib.read_metadata("sym1").metadata == results_dict[0].metadata
+    assert lib.read_metadata("sym2").metadata == results_dict[1].metadata
+    assert lib.read_metadata("sym3").metadata == results_dict[3].metadata
+
+
+def test_read_meta_batch_with_as_ofs(arctic_library):
+    lib = arctic_library
+    lib.write_pickle("sym1", 1, {"meta1": 1}, prune_previous_versions=False)
+    lib.write_pickle("sym2", 2, {"meta2": 2}, prune_previous_versions=False)
+    lib.write_pickle("sym3", 3, {"meta3": 3}, prune_previous_versions=False)
+
+    lib.write_pickle("sym1", 1, {"meta1": 4}, prune_previous_versions=False)
+    lib.write_pickle("sym2", 2, {"meta2": 5}, prune_previous_versions=False)
+    lib.write_pickle("sym3", 3, {"meta3": 6}, prune_previous_versions=False)
+
+    lib.write_pickle("sym1", 1, {"meta1": 7}, prune_previous_versions=False)
+    lib.write_pickle("sym2", 2, {"meta2": 8}, prune_previous_versions=False)
+    lib.write_pickle("sym3", 3, {"meta3": 9}, prune_previous_versions=False)
+    # v0 and v1 should be pruned by now.
+
+    results_dict = lib.read_metadata_batch(["sym1", "sym2", "sym_no_exist" ,"sym3"])
+    assert results_dict[0].metadata == {"meta1": 7}
+    assert results_dict[1].metadata == {"meta2": 8}
+    assert results_dict[2].metadata is None
+    assert results_dict[3].metadata == {"meta3": 9}
+
+    assert lib.read_metadata("sym1").metadata == results_dict[0].metadata
+    assert lib.read_metadata("sym2").metadata == results_dict[1].metadata
+    assert lib.read_metadata("sym3").metadata == results_dict[3].metadata
+
+
+    results_dict = lib.read_metadata_batch([ReadRequest("sym1", as_of=0), ReadRequest("sym2", as_of=0), ReadRequest("sym2", as_of=5), "sym_no_exist", ReadRequest("sym3", as_of=0)])
+    assert results_dict[0].metadata == {"meta1": 1}
+    assert results_dict[1].metadata == {"meta2": 2}
+    assert results_dict[2].metadata is None
+    assert results_dict[3].metadata is None
+    assert results_dict[4].metadata == {"meta3": 3}
+
+    results_dict = lib.read_metadata_batch([ReadRequest("sym1", as_of=1), ReadRequest("sym2", as_of=1), ReadRequest("sym2", as_of=5), "sym_no_exist", ReadRequest("sym3", as_of=1)])
+    assert results_dict[0].metadata == {"meta1": 4}
+    assert results_dict[1].metadata == {"meta2": 5}
+    assert results_dict[2].metadata is None
```

Comment:
The API isn't helping to disambiguate between "the version exists and there is no metadata" and "the version does not exist at all" (which I think is the case for "sym2" as_of=5

Reply:
I think that's the idea. We don't want to error the whole batch because a single version doesn't exist, so there are two cases where you get a None, when the Symbol/Version pair doesn't exist or when it does exist but doesn't have metadata. We could change the API but we'd need to make the case for the difference in (null) return value being useful to the end-user.